packages/react/src/components/uiComponents.tsx

```diff
@@ -49,10 +55,26 @@ type UserButtonExportType = typeof _UserButton & {
   MenuItems: typeof MenuItems;
   Action: typeof MenuAction;
   Link: typeof MenuLink;
+  /**
+   * The `Outlet` component can be used in conjunction with `asProvider` in order to control rendering
+   * of the OrganizationSwitcher without affecting its configuration or any custom pages
+   * that could be mounted
+   * @experimental This API is experimental and may change at any moment.
+   */
+  __experimental_Outlet: typeof UserButtonOutlet;
 };
 
-type UserButtonPropsWithoutCustomPages = Without<UserButtonProps, 'userProfileProps'> & {
+type UserButtonPropsWithoutCustomPages = Without<
+  UserButtonProps,
+  'userProfileProps' | '__experimental_asStandalone'
+> & {
   userProfileProps?: Pick<UserProfileProps, 'additionalOAuthScopes' | 'appearance'>;
+  /**
+   * Adding `asProvider` will defer rendering until the `Outlet` component is mounted.
+   * @experimental This API is experimental and may change at any moment.
+   * @default undefined
+   */
+  __experimental_asProvider?: boolean;
```

Comment:
I think you can make your life a little bit easier by making this `__experimental_asProvider: boolean` and then set `false` as a default.

Then you don't have to do `!!` or `?.` in other places to check for `undefined` or cast that to a boolean.

Reply:
This would be breaking right ? If I do that all UserButtons and OrgSwitchers would have to define asProvider which is not what we want

Reply:
Oh, I thought these would be our internal types, not public facing. If that's the case, then yeah, it needs to be optional of course